flake.nix

```diff
@@ -0,0 +1,78 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-24.11";
+    flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, pre-commit-hooks }:
+    flake-utils.lib.eachDefaultSystem
+      (system:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [
+              self.overlays.default
+            ];
+          };
+          hl = pkgs.haskell.lib;
+
+        in
+        {
+          packages = {
+            inherit (pkgs.haskellPackages) keter;
+            default = pkgs.haskellPackages.keter;
+          };
+
+          checks = {
+            inherit (pkgs.haskellPackages) keter;
+
+            pre-commit-check = pre-commit-hooks.lib.${system}.run {
+              src = ./.;
+              hooks = {
+                cabal-fmt.enable = true;
+                deadnix.enable = true;
+                hlint.enable = true;
+                markdownlint.enable = true;
+                nixpkgs-fmt.enable = true;
+                statix.enable = true;
+                stylish-haskell.enable = true;
+              };
+            };
```

Comment:
I'm not a fan of putting formatting checks in CI, but you're working on it so if it helps you go ahead.